.github/workflows/push_to_docker_hub.yml

```diff
@@ -0,0 +1,143 @@
+# This workflow replaces $SAGE_ROOT/.gitlab-ci.yml which has been
+# used formerly to make Docker images available on DockerHub.
+# The new workflow uses the same $SAGE_ROOT/docker/Dockerfile and
+# is based on docker/build-push-action@v4.
+#
+# A replacement has become neccessary because the GitLab runners
+# stopped to create new images since release 9.7.
+#
+# Concerning the images placed in the sagemath-dev repository you
+# should note the following difference to the releases build before
+# 9.8: They are no longer build on target sagemath-dev since this
+# causes a "No space left on device" error. Instead they are build
+# on target make-build, now. The main difference is the missing
+# documentation and that it enters in a bash shell.
+#
+# Sebastian Oehms, August 2023
+
+name: Build Docker images and push to DockerHub
+
+on:
+  workflow_dispatch:
+    # Allow to run manually
+    branches:
+      - 'develop'
+      - 'docker_hub_gha'
+  push:
+    tags:
+      # Just create image on pushing a tag
+      - '*'
+
+jobs:
+  sagemath-dev:
+    name: Build Docker image on target make-build and push to DockerHub sagemath-dev
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set tag
+        # docker/metadata-action@v4 is not used since we need to distinguish
+        # between latest and develop tags
+        id: set_tag
+        run: |
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+          TAG_NAME=$(git tag --sort=v:refname | tail -1)
+          TAG="sagemath/sagemath-dev:$TAG_NAME"
+          TAG_LIST="$TAG, sagemath/sagemath-dev:develop"
+          TAG_LIST="$TAG" # don't tag develop until meaning of sagemath-dev is clear
+          echo "TAG_NAME=$TAG_NAME" >> $GITHUB_ENV
+          echo "TAG=$TAG" >> $GITHUB_ENV
+          echo "TAG_LIST=$TAG_LIST" >> $GITHUB_ENV
+
+      - name: Update Tag List
+        id: upd_tag_list
+        run: |
+          TAG_LIST="${{ env.TAG_LIST }}, sagemath/sagemath-dev:latest"
+          TAG_LIST="${{ env.TAG_LIST }}"  # don't tag latest until meaning of sagemath-dev is clear
+          echo "TAG_LIST=$TAG_LIST" >> $GITHUB_ENV
+        if: "!contains(env.TAG_NAME, 'beta') && !contains(env.TAG_NAME, 'rc')"
+
+      - name: Check env
+        run: |
+          echo ${{ env.TAG_NAME }}
+          echo ${{ env.TAG }}
+          echo ${{ env.TAG_LIST }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
```

Comment:
Why do we need QEMU here and below?

Reply:
> Why do we need QEMU here and below?

It's there from the master I used. I will drop it in the next commit.